src/hakunapi-jsonfg/src/main/java/fi/nls/hakunapi/geojson/hakuna/JSONFGFeatureCollectionWriter.java

```diff
@@ -0,0 +1,233 @@
+package fi.nls.hakunapi.geojson.hakuna;
+
+// TODO: forced to share package fi.nls.hakunapi.geojson.hakuna due to json.writeStringUnsafe scope
+// TODO: multiple datetimeProperties
+// TODO: WGS84 check spec
+// TODO: figure out when to write geometry and/or place with or without coordRefSys
+// TODO: preprojected geoms?
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.List;
+
+import org.locationtech.jts.geom.Geometry;
+
+import fi.nls.hakunapi.core.FeatureType;
+import fi.nls.hakunapi.core.FloatingPointFormatter;
+import fi.nls.hakunapi.core.GeometryWriter;
+import fi.nls.hakunapi.core.geom.HakunaGeometry;
+import fi.nls.hakunapi.core.geom.HakunaGeometryDimension;
+import fi.nls.hakunapi.core.geom.HakunaGeometryJTS;
+import fi.nls.hakunapi.core.projection.NOPProjectionTransformer;
+import fi.nls.hakunapi.core.projection.ProjectionHelper;
+import fi.nls.hakunapi.core.projection.ProjectionTransformer;
+import fi.nls.hakunapi.core.property.simple.HakunaPropertyGeometry;
+import fi.nls.hakunapi.core.schemas.Crs;
+import fi.nls.hakunapi.core.schemas.Link;
+import fi.nls.hakunapi.core.util.CrsUtil;
+import fi.nls.hakunapi.jsonfg.JSONFG;
+
+public class JSONFGFeatureCollectionWriter extends HakunaGeoJSONFeatureCollectionWriter implements JSONFG {
+
+	protected int outputSrid;
+	protected JSONFGGeometryWriter placeJson;
+
+	// local cached (+inherited geometryCached & cachedGeometry )
+	FeatureType collectionFt = null;
+	LocalDate date = null;
+	Instant timestamp = null;
+	ProjectionTransformer outputWgs84Proj;
+
+	@Override
+	public void init(OutputStream out, FloatingPointFormatter formatter, int srid, boolean crsIsLatLon)
+			throws IOException {
+		super.init(out, formatter, Crs.CRS84_SRID, crsIsLatLon);
+		outputSrid = srid;
+
+	}
+
+	@Override
+	public void initGeometryWriter(HakunaGeometryDimension dims) {
+		this.dims = dims;
+		geometryJson = new JSONFGGeometryWriter(json, HakunaGeoJSON.GEOMETRY, true, dims);
+		placeJson = new JSONFGGeometryWriter(json, PLACE, true, dims);
+		propertyGeometryJson.clear();
+	}
+
+	protected void writeFeatureCollection() throws IOException {
+	}
+
+	@Override
+	public void writeProperty(String name, HakunaGeometry geometry) throws Exception {
+		openProperties();
+		final GeometryWriter geomWriter = propertyGeometryJson.computeIfAbsent(name, k -> new JSONFGGeometryWriter(json,
+				k.getBytes(StandardCharsets.UTF_8), forceLonLat || !crsIsLatLon, dims));
+		geometry.write(geomWriter);
+	}
+
+	@Override
+	public void startFeatureCollection(FeatureType ft, String layername) throws Exception {
+
+		collectionFt = ft;
+		if (isWGS84(outputSrid)) {
```

Comment:
I guess we could just leave it up to ProjectionTransformerFactory to determine if toCRS84(srid) is NOP? That way JSONFG doesn't have to care about 84, 4326 or 4258 being WGS84.

Reply:
retained srid==84 check for now NOP check did not work